### ISSUE TITLE
[REQ] Enhance Gradle plugin to support template locations on the classpath

### ISSUE THREAD

**Issue by dimber-cais** [OpenAPITools/openapi-generator#14678]
### Is your feature request related to a problem? Please describe.

I have set of customized templates that I wish to apply across all of my projects (dozens). Currently I need to supply those files as files within a file-system directory (via `templateDir`) which isn't ideal given I have so many projects. 

## Describe the solution you'd like

I would like a new parameter added to the Gradle plugin, to specify a classpath location for templates. Using this approach, I can add the templates as a gradle `compileOnly` dependency.

## Describe alternatives you've considered

As a workaround, I can write a gradle task that executes before the `GenerateTask` which copies the files from the Jar into a file-system directory, which is then configured as the `templateDir`.

**Comment by Bryksin** [comment]
The same problem.
According to our tests in the plugin version `6.1.0` - templates were working from the classpath
but in `6.3.0` - does not anymore, though in 6.3.0 added feature of `remoteInputSpec` and `auth` to grab the spec directly from SwaggerHub - which is very handy!

As a workaround, I included a jar with templates as dependencies and extracted them into the local dir in the custom task

here is how it works:

```
configurations {
	implementation {
		canBeResolved = true
	}
}

tasks.register("extractTemplates", Copy) {
	from zipTree("${rootProject.configurations.implementation.find { it.name.contains('swagger-codegen-templates') }}")
	into "${rootDir}"
	exclude "META-INF/**"
}

tasks.compileJava.dependsOn tasks.openApiGenerate
tasks.openApiGenerate.dependsOn tasks.extractTemplates

```

but the downside is that the custom task `extractTemplates` is never `UP-TO-DATE` on any next build.
So basically build happening every time whenever we trigger any gradle task (for example with sonar qube scan or publishing to Artifactory)

**Comment by dimber-cais** [comment]
This might be as simple as adding a new property to `GenerateTask` as:

```kotlin
    /**
     * Resource path containing template files.
     */
    @Optional
    @Input
    val templateResourcePath = project.objects.property<String?>()
```

Then in `GenerateTask.doWork()` add:

```kotlin
            templateDir.ifNotEmpty { value ->
                configurator.setTemplateDir(value)
            }

            templateResourcePath.ifNotEmpty { value ->
                if (null != templateDir) {
                    logger.warn("Both templateDirectory and templateResourcePath were configured. templateResourcePath overwrites templateDirectory.")
                }
                configurator.setTemplateDir(value)
            }
```

This seems to be exactly how the Maven plugin achieves the same functionality requested in this issue.

**Comment by dimber-cais** [comment]
This is now covered in a PR: https://github.com/OpenAPITools/openapi-generator/pull/14909

I've built this locally and integrated and it works well. One thing to consider, is that the plugin needs access to the JAR that contains the template. So you need something like this:

```kotlin
buildscript {
    dependencies {
        classpath("com.example.customized-openapi-templates:1.2.3")
    }
}
```
